research_daps/orms/glass.py

```diff
@@ -0,0 +1,202 @@
+    is_active = Column(
+        Boolean,
+        nullable=False,
+        index=True,
+        doc="Indicates whether `address_id` is active",
+    )
+    # TODO : should this really have two date columns: valid_from and valid_to ?
+    date = Column(
+        Date,
+        nullable=False,
+        index=True,
+        doc="Date of data-dump associating org with address",
+    )
+    organisation = relationship("Organisation", back_populates="addresses")
+    address = relationship("Address", back_populates="organisations")
+
+
+organisation_sector = Table(
+    "organisation_sector",
+    Base.metadata,
+    Column("org_id", Integer, ForeignKey("organisation.org_id")),
+    Column("sector_id", Integer, ForeignKey("sector.sector_id")),
+    # TODO: Active? boolean column indicating whether sector is in latest data dump
+    # TODO: Rank Integer column indicating sector rank
+    # TODO: Date Date column indicating date of latest dump with this relationship
+)
+
+notice_terms = Table(
+    "notice_terms",
+    Base.metadata,
+    Column("notice_id", Integer, ForeignKey("notice.notice_id")),
+    Column("term_id", Integer, ForeignKey("covid_term.term_id")),
+)
+
+
+class Organisation(Base):
+    """An organisation relating to a business website"""
+
+    org_id = Column(Integer, primary_key=True)
+    name = Column(
+        String,
+        unique=True,
+        doc="Organisation name inferred by named entity recognition",
+        index=True,
+    )
+    website = Column(String, nullable=False, doc="URL of organisation")
+    active = Column(
+        Boolean,
+        nullable=False,
+        doc="True if Organisation was contained in last data-dump",
+    )
+    # One-to-many
+    notices = relationship("Notice", backref=backref("organisation"))
+    descriptions = relationship(
+        "OrganisationDescription", backref=backref("organisation")
+    )
+    metadata_ = relationship("OrganisationMetadata", backref=backref("organisation"))
+    # Many-to-many
+    sectors = relationship(
+        "Sector", secondary=organisation_sector, back_populates="organisations"
+    )
+    addresses = relationship("OrganisationAddress", back_populates="organisation")
+    # addresses = relationship(
+    #     "Address", secondary=organisation_address, back_populates="organisations"
+    # )
+
+
+class OrganisationMetadata(Base):
+    """Organisation metadata which may not be stable over time"""
+
+    org_id = Column(Integer, ForeignKey("organisation.org_id"), primary_key=True)
+    date = Column(
+        Date, primary_key=True, doc="Date of data-dump inserting row", index=True
+    )
+    has_webshop = Column(
+        Boolean, nullable=False, doc="If True, presence of a webshop was found"
+    )
+    vat_number = Column(String, doc="VAT number (low population)")
+    low_quality = Column(
+        Boolean,
+        nullable=False,
+        doc="If True, information for `org_id` was of low quality",
+    )
+
+
+class OrganisationDescription(Base):
```

Comment:
Is the idea here that a) Glass already provide multiple descriptions or b) You are intending to update with multiple descriptions over time? If b) I think the up{sert/dat}ing logic isn't consistent across the ORMs but if a) then fine!